Todo/TaskListsViewController.swift

```diff
@@ -0,0 +1,271 @@
+                let doc = self.listRows![indexPath.row].document!
+                
+                Ui.showTextInputDialog(
+                    onController: self,
+                    withTitle: "Edit List",
+                    withMessage:  nil,
+                    withTextFieldConfig: { (textField) -> Void in
+                        textField.placeholder = "List name"
+                        textField.text = doc["name"] as? String
+                        textField.autocapitalizationType = .Words
+                    },
+                    onOk: { (name) -> Void in
+                        self.updateTaskList(doc, withName: name)
+                    }
+                )
+            }
+            update.backgroundColor = UIColor(red: 0.0, green: 0.48, blue: 1.0, alpha: 1.0)
+            
+            return [delete, update]
+    }
+    
+    // MARK: - UISearchController
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let text = searchController.searchBar.text ?? ""
+        let active = !text.isEmpty
+        listsLiveQuery.startKey = active ? text : nil
+        listsLiveQuery.endKey = listsLiveQuery.startKey
+        listsLiveQuery.prefixMatchLevel = active ? 1 : 0
+        listsLiveQuery.queryOptionsChanged()
+    }
+    
+    // MARK: - Action
+    @IBAction func addAction(sender: AnyObject) {
+        Ui.showTextInputDialog(
+            onController: self,
+            withTitle: "New Task List",
+            withMessage:  nil,
+            withTextFieldConfig: { (textField) -> Void in
+                textField.placeholder = "List name"
+                textField.autocapitalizationType = .Words
+            },
+            onOk: { (name) -> Void in
+                self.createTaskList(name)
+            }
+        )
+    }
+    
+    @IBAction func logoutAction(sender: AnyObject) {
+        let app = UIApplication.sharedApplication().delegate as! AppDelegate
+        app.logout()
+    }
+
+    // MARK: - Database
+
+    func setupViewAndQuery() {
+        let listsView = database.viewNamed("list/listsByName")
+        if listsView.mapBlock == nil {
+            listsView.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task-list" {
+                    if let name = doc["name"] {
+                        emit(name, nil)
+                    }
+                }
+            }, version: "1.0")
+        }
+
+        listsLiveQuery = listsView.createQuery().asLiveQuery()
+        listsLiveQuery.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        listsLiveQuery.start()
+        
+        let incompTasksCountView = database.viewNamed("list/incompleteTasksCount")
+        if incompTasksCountView.mapBlock == nil {
+            incompTasksCountView.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task" {
```

Comment:
You can use `,` to combine multiple `if let`s, instead of getting this multiply-indented block.